### DISCUSSION TITLE
Make reejs the default js bundler for non-js backends - add Documentation about how to do that

### DISCUSSION THREAD

**Discussion by schlichtanders** [rovelstars/reejs#29] [Ideas]
Hi there, I am using another language as the backend and like reejs approach to marry npm dependency management with import maps. This way dependencies can be controlled, while still using vanilla javascript/html/css.

A workflow where the files itself are the dev version becomes possible. Of course with this you cannot use jsx or ts files which need to be precompiled first, but I am very happy to use plain js files with TypeScript annotations.

Unfortunately, reejs has its own opinion about how the folder structure shall look like, which is a deal breaker for me, as src directory is already occupied by my backend programming language.

It would be great to simply have a couple of config examples, how to use reejs for such purposes where the files themselves represent the dev version and reejs is only used for bundling everything so that no reference to esm.sh is left for production stability.

**Comment by schlichtanders** [comment]
After further research, I found jspm.org which is more towards my need of having a importmaps focused package management.
I will probably try that instead.

**Reply by renhiyama** [reply]
Whatever works for you 😀

**Reply by renhiyama** [reply]
Reejs saves downloaded dependencies in `.reejs` folder.
As for the "rules" that reejs creates, you can easily cancel them out - packit is the one creating those rules, and packit allows you to modify its internal writers and readers.

**Reply by renhiyama** [reply]
https://ree.js.org/docs/basics/packit <- tutorial on how packit works.

Here's the default code that packit injects into itself. You can freely replace any existing packit reader/writer/etc, with your own implementation.

https://github.com/rovelstars/reejs/blob/main/src/cli/cmds/utils/Packit.js

**Reply by renhiyama** [reply]
@schlichtanders

**Reply by schlichtanders** [reply]
Thank you a lot for the pointers. The next days I am going to further try to to importmaps.
Thank you for the work on reejs

**Reply by renhiyama** [reply]
Your welcome! If you like reejs, feel free to star it, or make an donation to keep me motivated!
